### DISCUSSION TITLE
Clarification on Grid-Stat tool's input grids

### DISCUSSION THREAD

**Discussion by gsketefian** [dtcenter/METplus#1764] [File I/O] [answered]
Hi,

I am trying to clarify whether the forecast and observations passed to Grid-Stat must be on the same grid or may be on different grids.  The documentation [here](https://met.readthedocs.io/en/latest/Users_Guide/grid-stat.html) says:

> The Grid-Stat tool provides verification statistics for a matched forecast and observation grid. All of the forecast grid points in the region of interest are matched to observation grid points on the same grid. All the matched grid points are used to compute the verification statistics. The Grid-Stat tool functions in much the same way as the Point-Stat tool, except that no interpolation is required because the forecasts and observations are on the same grid. 

Others in the DTC ensemble group I'm working in said Grid-Stat does in fact do interpolation, but the description above seems to not agree with that assessment ("no interpolation is required because the forecasts and observations are on the same grid").  Just wondering which is correct, and whether the documentation needs clarification or an update.  Thanks.

**Comment by j-opatz** [comment]
Hello,

And thank you for your question. You've identified what I think is a less than ideally worded section of MET's user's guide: the grid-stat tool can absolutely handle differing grid types between the observation and forecast fields. Doing so will require the use of the `regrid` dictionary, described [here](https://met.readthedocs.io/en/latest/Users_Guide/config_options.html#regrid). You can also provide the tool with fields that are on the same grid and can skip the regrid requirement, although it is still an option if there is a secondary grid you'd like the data verified on. 

What I believe this entry was trying to convey is that if the fields are on the same grid, there is no regridding/interpolation needed. Whereas with the use of point-stat, point data will always require some type of interpolation to fit on to a chosen grid.

Would a better description be

> The Grid-Stat tool functions in much the same way as the Point-Stat tool, except that no interpolation is required **when** the forecasts and observations are on the same grid

**Comment by gsketefian** [comment]
Hi @j-opatz, 

Thanks for the clarification.  Happy to hear Grid-Stat can interpolate between different grids.  So it's a matter of clarifying that paragraph, as you say.

In my opinion, the sentence

> All of the forecast grid points in the region of interest are matched to observation grid points on the same grid. 

is also misleading.  I would rewrite that whole introductory paragraph to make it clear that the forcecast and obs don't have to be on the same grid and that regridding is performed as necessary.  I've included my suggested version below, please feel free to use as much or as little of it as you see fit.  Thanks.

> The Grid-Stat tool functions in much the same way as the Point-Stat tool except that the verification statistics it calculates are for a matched forecast and observation grid (as opposed to a set of observation points). The user-provided forecast and observation grids may be different, in which case the forecast grid points are matched (regridded/interpolated) to the observation grid points [is it possible to go the other way, i.e. regrid the obs to the forecast grid?  If yes, please clarify/reword]. If the input forecast and observation grids are identical, then no regridding is necessary. The matched forecast-observation pairs on the observation grid are then used to compute the verification statistics.

**Comment by gsketefian** [comment]
Hi @j-opatz, 

I just read the “to_grid” entry in the regrid dictionary, and now I understand that there are in general 3 grids.  So I’ve reworded my suggested paragraph:

> The Grid-Stat tool functions in much the same way as the Point-Stat tool except that the verification statistics it calculates are for a matched forecast-observation grid (as opposed to a set of observation points). Neither the forecast nor the observation grid needs to be identical to the matched grid (i.e. the grid on which the forecast and observations are matched into pairs).  If the forecast grid is different than the matched grid, then forecast values are regridded (interpolated) to the matched grid.  The same is done for observations if necessary.  If the forecast and observation grids are identical to the matched grid, then no regridding is performed. The forecast-observation pairs on the matched grid are then used to compute the verification statistics.

Thanks again.
Gerard

**Reply by j-opatz** [reply]
Thanks Gerard,

I'm going to run this rewrite past the team in our next meeting and gather input on the suggested improvements you've provided. Hopefully we can come to an agreement on how to proceed and you'll have a more clear path forward using this section of the User's Guide next time.

**Comment by gsketefian** [comment]
@j-opatz Sounds good.  Whatever the team decides is good with me :)

**Reply by j-opatz** [reply]
Hi Gerard,

Take a look at the rewrite that was based on your suggestions and let me know if it provides a more clear description of the Grid-Stat tool. We can fine tune anything that still seems unclear and hopefully successfully resolve this Discussions.

https://github.com/dtcenter/MET/blob/develop/docs/Users_Guide/grid-stat.rst#introduction

**Comment by gsketefian** [comment]
Hi John,
I made a couple of suggestions/modifications to the intro in [this](https://docs.google.com/document/d/1uM4aaJ1965VA-hptWUt1QDtmv3OnFeVaeKLpN27xo2k/edit) google doc (you should have edit permission via your NOAA account).  Feel free to use as you see fit.

Btw, I only read the intro paragraph assuming that's where all the changes are.  Please let me know if you'd like me to look over any other parts.  Thanks.
Gerard

**Reply by j-opatz** [reply]
Gerard,

I've gone ahead and reviewed the changes, added some comments and request one more direction on verbiage from you. Hopefully I can get the final writing in and up for the next beta on Wednesday!

**Comment by gsketefian** [comment, marked as answer]
@j-opatz I just finished replying to the comments in the [google doc](https://docs.google.com/document/d/1uM4aaJ1965VA-hptWUt1QDtmv3OnFeVaeKLpN27xo2k/edit).  Please let me know if you have questions.  Thanks for asking for my feedback.

**Reply by j-opatz** [reply]
@gsketefian,

I hope that everything is going well and that the changes to the documentation were a step in the right direction for users. We had discussed that an additional part of this Discussion is a location to clearly state the order of regridding and smoothing operations (possibly more, but I'm not remembering). Any new thoughts on where that would be best suited?

**Comment by gsketefian** [comment]
Hi John,
Sorry I haven't gotten to this yet.  Working through my to-do list.  I will
likely get to it sometime next week.  Thanks.
Gerard

On Tue, Sep 20, 2022 at 11:16 AM j-opatz ***@***.***> wrote:

> @gsketefian <https://github.com/gsketefian>,
>
> I hope that everything is going well and that the changes to the
> documentation were a step in the right direction for users. We had
> discussed that an additional part of this Discussion is a location to
> clearly state the order of regridding and smoothing operations (possibly
> more, but I'm not remembering). Any new thoughts on where that would be
> best suited?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/dtcenter/METplus/discussions/1764#discussioncomment-3692684>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AHM3ZYQJGM4NYEO7ARHDUPTV7HWP5ANCNFSM57UKTVUQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>


-- 
Gerard Ketefian
Research Scientist
NOAA/OAR/ESRL/GSD/EMB, R/GSD1
325 Broadway
Boulder, CO 80305
phone: 303-497-6209

**Comment by j-opatz** [comment]
Hi Gerard,

It's been some time since this discussion was last touched; were you able to provide any more thoughts on the best location for the information on regridding and smoothing in the MET user's guide? If so, we can pick this back up and try to determine the computational order they're completed in. With that in hand, your guidance on a write up for the information could be captured and we can try to get that into the guide.

**Reply by j-opatz** [reply]
Given the length of time since the last comment from the original poster, the Discussion is being closed for inactivity. If new questions arise, please open a new Discussion and we'll respond there.
